core/Repository/Credentials/PortalRepositoryCredentials.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Cmf.CLI.Core.Constants;
+using Cmf.CLI.Core.Enums;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cmf.CLI.Core.Repository.Credentials
+{
+    public class PortalRepositoryCredentials : IRepositoryCredentials, IRepositoryInteractiveLogin, IRepositoryDerivedCredentials
+    {
+        #region Constants
+
+        public const string TokenEnvVar = "CM_PORTAL_TOKEN";
+        public static readonly string TokenDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData, Environment.SpecialFolderOption.Create), "cmfportal");
+        public static readonly string TokenFilePath = Path.Combine(TokenDir, "cmfportaltoken");
+
+        #endregion Constants
+
+        protected IFileSystem _fileSystem;
+
+        public PortalRepositoryCredentials(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+        }
+
+        public AuthType[] SupportedAuthTypes => [AuthType.Bearer];
+
+        public RepositoryCredentialsType RepositoryType => RepositoryCredentialsType.Portal;
+
+        public PropertyRequirement KeyPropertyRequirement => PropertyRequirement.Ignored;
+
+        public PropertyRequirement DomainPropertyRequirement => PropertyRequirement.Ignored;
+
+        public void ValidateCredentials(IList<ICredential> credentials)
+        { }
+
+        public Task SyncCredentials(IList<ICredential> credentials)
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task<ICredential> InteractiveLogin()
```

Comment:
I also kinda don't like the name here, it seems we're supporting interactive login ourselves, but there was no mention of the AnsiConsole methods we use for this.

Reply:
Renamed the class and the method to remove references to `Interactive`, chose `AutomaticLogin` instead.